climate-monitoring-core/src/main/java/climatemonitoring/core/Result.java

```
@@ -1,6 +1,24 @@
 package climatemonitoring.core;
 
-public abstract class Result {
+public abstract class Result<T> {
 
-	
+	public Result() {
+
+
+	}
+
+	public abstract void exec();
+
+	public boolean ready() {
+
+		return !m_thread.isAlive();
+	}
+
+	public T get() {
+
+		return m_returnData;
+	}
```

This function should throw an exception. Possible implementation
```
public T get() throws Exception {

	if (m_exception != null) throw m_exception;

	return m_returnData;
}
```

---

Fixed